### ISSUE TITLE
Connecting 2nd controller  results disconnection 1st one

### ISSUE THREAD

**Issue by yarodevuci** [Ryochan7/DS4Windows#2458]
I am not 100% sure if that's a bug or no but here is my steps.
1. I connected 1st PS4 controller. It appeared in the DS4
2. I connected 2nd PS5 controller. It appeared in DS4 as Controller 2nd

Few seconds later Controller 1st gets disconnected. So I am not able to have 2 controller at the same time. 

I have tried connecting Kensington Bluetooth USB Dongle; however, DS4 does not see the controller. 

I am using a PC with MSI Godlike Motherboard. Turning the WiFi OFF did not do anything. 

Here some Logs:

```
07/09/2022 00:47:16: The gamepad DS4 v.2 (BT) uses custom feature set (MonitorAudio)
07/09/2022 00:47:16: Found Controller: 4C:B9:9B:06:D9:1A (BT) (DS4 v.2)
07/09/2022 00:47:16: Plugging in virtual X360 Controller
07/09/2022 00:47:16: Associate X360 Controller in slot #1 for input DS4 v.2 controller #1
07/09/2022 00:47:16: Controller 1 is using Profile "Default". Battery level is 87%
07/09/2022 01:02:46: Found Controller: D0:BC:C1:D6:71:56 (BT) (DualSense)
07/09/2022 01:02:46: Plugging in virtual X360 Controller
07/09/2022 01:02:46: Associate X360 Controller in slot #2 for input DualSense controller #2
07/09/2022 01:02:46: Controller 2 is using Profile "Default". Battery level is 75%
07/09/2022 01:02:49: 4C:B9:9B:06:D9:1A disconnected due to read failure: 997
07/09/2022 01:02:49: Gamepad data write connection is lost. Disconnecting the gamepad. LastErrorCode=0
07/09/2022 01:02:49: Disassociate X360 Controller from slot #1 for input DS4 v.2 controller #1
07/09/2022 01:02:49: Unplugging virtual X360 Controller
07/09/2022 01:02:49: Controller 1 was removed or lost connection
07/09/2022 01:18:14: D0:BC:C1:D6:71:56 disconnecting due to idle disconnect
07/09/2022 01:18:14: Disassociate X360 Controller from slot #2 for input DualSense controller #2
07/09/2022 01:18:14: Unplugging virtual X360 Controller
07/09/2022 01:18:14: Controller 2 was removed or lost connection
07/09/2022 01:19:00: Found Controller: D0:BC:C1:D6:71:56 (BT) (DualSense)
07/09/2022 01:19:00: Plugging in virtual X360 Controller
07/09/2022 01:19:00: Associate X360 Controller in slot #1 for input DualSense controller #1

```

Please help.

**Comment by Masamune3210** [comment]
Try with USB, could be your Bluetooth chipset just can't handle two controllers worth of data at once, especially since DS4 and DualSense both use up more bandwidth than a typical controller would due to light bar, rumble, and motion all talking at once

**Comment by yarodevuci** [comment]
@Masamune3210 USB is not an option since I am streaming to a TV.

**Comment by Ryochan7** [comment]
Did you have Steam running at the time? Even if you do not have PlayStation Configuration Support enabled for Steam Input, Steam can do some init sequence for the DS4 and DualSense if it detects a controller. That could cause some odd behavior if both apps try to perform a mode change on the controller around the same time.

**Comment by yarodevuci** [comment]
@Ryochan7 no steam process was running at that time.

**Comment by Trihy** [comment]
2 x PS4 controllers could be a better option. They take less BW than one dualsense. Also intel BT adapters are bad at handling more than one controller.

**Comment by jydhthrshrhty** [comment]
This is a classic bug with hidhide not configured properly

**Comment by jydhthrshrhty** [comment]
> 2 x PS4 controllers could be a better option. They take less BW than one dualsense. Also intel BT adapters are bad at handling more than one controller.

not true at all we have played here 4-8 controller with a ax210 at the same time here

**Comment by Trihy** [comment]
That's not possible with that BT adapter. Maybe PS4 controllers with disabled send output data.

With dualsense more than one is not stable unless you disable send output data.

**Comment by AsheX1** [comment]
> 

i can confirm this 2x intel BL is bad it can do 1 controller DS4 or 5 but as soon as i hook up 2 controllers either 2 DS4 or 5 one of the 2 are gone get disconnect at random in a 1 min time frame somewhere

**Comment by AsheX1** [comment]
some dit tell me something like a few days ago to turn off power saving  in (Device Manager)
looking at it now
Device Manager > Human Interface Devices > anything BL related  if it has a power management tab

i may need to re-test my controllers to see if it can hanlde 2 or maybe even 3,4

**Comment by Trihy** [comment]
Hi. Not sure about that. Dont think it will help that much. Must be a driver or hw limitation.

Another thing this bt adapters cant do is connecting one dualsense and one bt headset. You have to connect the bt headset before dualsense or wont connect at all. And if you stop the audio stream, headset will disconnect and wont reconnect until you turn off the duslsense. A really great bt adapter. lol.

**Comment by AsheX1** [comment]
wel after a quick test my 2 PC can now do 2 PS5 controllers 
Device Manager > Human Interface Devices > just check everything and disable power saving on everything

**Comment by Trihy** [comment]
Hi. Try to be more specific, so we can make some test. Also if you have a bt headset, try it too. Thanks

**Comment by AsheX1** [comment]
gone need to check for more stuff to hook up wish i knew where my PS4 controllers are right now

**Comment by Trihy** [comment]
Sure, try to do some test when you have some time and report findings.

**Comment by AsheX1** [comment]
once i have some more time and am home again im gone test some more kinda busy IRL work stuff on the road etc i have not forgot about this just a reminder

**Comment by Trihy** [comment]
Sure, no problem. Sometimes I can get both to work, but without changing any setting.

**Comment by CaptainM19** [comment]
I've got the same issue. I can connect Dualsense and Dualshock 4 to my laptop just fora  few seconds and one of these is suddenly loosing connection. Can You help me, because i made everything what @jydhthrshrhty said and the situation is still the same? My OS: Windows 11 21H2

**Comment by Trihy** [comment]
> I've got the same issue. I can connect Dualsense and Dualshock 4 to my laptop just fora few seconds and one of these is suddenly loosing connection. Can You help me, because i made everything what @jydhthrshrhty said and the situation is still the same? My OS: Windows 11 21H2

The only solution atm is getting a better BT adapter.

**Comment by sergiu99** [comment]
to confirm, on my end cannot handle PS5 and PS4 controller together with Intel AX210

**Comment by Trihy** [comment]
There are threads on Intel wireless forum about this problem. Add your experience to them so they cant ignore us anymore.

**Comment by Tristan-2i** [comment]
I have the same problem. When I connect two controllers (dualshock 4 and 5) in bluetooth no problem but as soon as I start ds4 Windows one of the two controllers disconnects (the dualshock 4).
No problem with only one controller connected and no problem without ds4 windows.
I have tried different fixes and I have HidHide configured. Has anyone found a solution?

**Comment by Trihy** [comment]
There is no fix. You have to ask for a fix to your BT adapter manufacturer. Or buy other.

**Comment by Tristan-2i** [comment]
> Which BT adapter do you have?

I have a legion 5 pro laptop so I can't realy changer the BT adapter.
Is it really a BT adapter problem since everything works fine without ds4 windows?

**Comment by Trihy** [comment]
Intel AX BT cards support 1 and a half dualsense.

The half means sometimes the second controller could stay connected for a while, sometimes just a few seconds. It doesn't support 2 controllers at the same time without disconnects.

Buy a BT dongle or ask Intel for a fix.

**Comment by Tristan-2i** [comment]
> Intel AX BT cards support 1 and a half dualsense.
> 
> The half means sometimes the second controller could stay connected for a while, sometimes just a few seconds. It doesn't support 2 dualsense or 1 dualsense + other controllers at the same time without disconnects.
> 
> Buy a BT dongle or report the problem to intel to see if they can help.

My old legion y520 laptop can easly connect multiples controllers and my new one can't ?

**Comment by Trihy** [comment]
Sure. It's a problem, limitation, driver problem, firmware, etc of Intel AX BT cards.

**Comment by Tristan-2i** [comment]
> Sure. It's a problem, limitation, driver, firmware, etc of Intel AX BT cards.

But why is there a problem only with ds4 windows?
Is it possible to fix it by installing another driver ?

**Comment by Trihy** [comment]
It's not a problem with ds4. It's a know problem and reported several times. It's an Intel problem. Sure you can try other drivers, some older ones works better.

**Comment by Tristan-2i** [comment]
I've been looking for a solution for a long time now and I don't see much talk about this subject (only one post on intel and very few on google)
So there is no solution for the moment?

**Comment by Trihy** [comment]
Most users uses just one controller. Do there won't be many reports.

Only solution is stop buying faulty products. Intel don't give a f as they keep selling.

**Comment by Ryochan7** [comment]
I can connect 3 devices at once with my BT 4.0 dongle without any issues (1 DS4 v.2, 1 DualSense, 1 Switch Pro). Not much can be done if there is something wrong up the chain like with the Bluetooth hardware or the Bluetooth software stack being used.

Comparing vanilla behavior (DInput mode) to using DS4Windows is not a valid comparison due to the mode switch that takes place. Better to compare behavior in DS4Windows to Steam Input as both will perform the mode switch to get the additional data from the controller (gyro, touchpad).

**Comment by Tristan-2i** [comment]
> I can connect 3 devices at once with my BT 4.0 dongle without any issues (1 DS4 v.2, 1 DualSense, 1 Switch Pro). Not much can be done if there is something wrong up the chain like with the Bluetooth hardware or the Bluetooth software stack being used.
> 
> Comparing vanilla behavior (DInput mode) to using DS4Windows is not a valid comparison due to the mode switch that takes place. Better to compare behavior in DS4Windows to Steam Input as both will perform the mode switch to get the additional data from the controller (gyro, touchpad). Also, getting the faster input poll rate which makes a big difference.

Indeed the problem happens also with steam, even on windows without having launched anything

**Comment by jydhthrshrhty** [comment]
What is your Bluetooth peripheral. Sent from Proton Mail for iOS On Mon, Dec 19, 2022 at 15:12, Tristan-2i ***@***.***> wrote:  

I can connect 3 devices at once with my BT 4.0 dongle without any issues (1 DS4 v.2, 1 DualSense, 1 Switch Pro). Not much can be done if there is something wrong up the chain like with the Bluetooth hardware or the Bluetooth software stack being used.
Comparing vanilla behavior (DInput mode) to using DS4Windows is not a valid comparison due to the mode switch that takes place. Better to compare behavior in DS4Windows to Steam Input as both will perform the mode switch to get the additional data from the controller (gyro, touchpad). Also, getting the faster input poll rate which makes a big difference.

Indeed the problem happens also with steam, even on windows without having launched anything

**Comment by Trihy** [comment]
Ryo, which BT dongle are you using? So people can buy it or look something similar.

Many works fine. But dualsense is a little BW hungry.

Besides Intel lame adapters. I have some generic cheap ones, that once paired and connected with a dualsense, they can't even find other devices for pairing.

**Comment by Masamune3210** [comment]
I'm not Ryo, but with both my DualSense and my DS4 connected at the same time and both vibrating as a quick test, I had no issues with this one
https://www.amazon.com/Bluetooth-Receiver-Keyboards-Controllers-USB-BT400/dp/B00DJ83070

**Comment by Tristan-2i** [comment]
I'm really interested in finding a BT dongle that supports up to 4 controllers or more if I can't find a solution with my built-in bluetooth

**Comment by Tristan-2i** [comment]
If anyone is interested, I'm continuing my research and it looks like a bluetooth driver problem since version 21.0.0.4

I have opened a thread on the intel forum:
https://community.intel.com/t5/Wireless/bluetooth-disconnection-with-multiple-controllers/td-p/1439726

I have opened a thread on the Lenovo forum:
https://forums.lenovo.com/t5/Gaming-Laptops/bluetooth-disconnection-with-multiple-controllers/m-p/5190280

And an old topic mentions the same problem:
https://community.intel.com/t5/Wireless/Can-Intel-AC-9260-Bluetooth-work-with-4-concurrent-PlayStation/td-p/662264

**Comment by EliasBey** [comment]
Hey guys, i may have found a solution to our problem.

I suspected the issue to be the intel card not recognizing the controllers being individual and independent so what i did was going to control pannel then device and printers (it may not be the same name, i'm sry i don't have my pc at hand) then you select the wireless controlers ( you should have one at least) and you go the window bluetooth there you should be able to double click on Wireless Controller and change its name to whatever you want (it is a way to show the card that you're not trying to connect the same controller again and again) once you change that you click on apply and you can try to reconnect your other controller / it may be the same procedure for many controllers you only have to renamme more i guess.

So that's how it worked for me hope it works for you too, and also if you can just copy paste my answer on the threads above in order to help more people as i don't have accounts on those websites.

**Comment by Tristan-2i** [comment]
> Hey guys, i may have found a solution to our problem.
> 
> I suspected the issue to be the intel card not recognizing the controllers being individual and independent so what i did was going to control pannel then device and printers (it may not be the same name, i'm sry i don't have my pc at hand) then you select the wireless controlers ( you should have one at least) and you go the window bluetooth there you should be able to double click on Wireless Controller and change its name to whatever you want (it is a way to show the card that you're not trying to connect the same controller again and again) once you change that you click on apply and you can try to reconnect your other controller / it may be the same procedure for many controllers you only have to renamme more i guess.
> 
> So that's how it worked for me hope it works for you too, and also if you can just copy paste my answer on the threads above in order to help more people as i don't have accounts on those websites.

I just tested and it doesn't seem to work for me with DS4Windows enabled to map to xbox 360 the second one disconnects

**Comment by Trihy** [comment]
Since I could find some dongles to try, started a compatibility list.

https://github.com/Trihy/BT-Dongle-Compatibility-List

Report findings and will add them.

**Comment by fcuadros** [comment]
I just bought a brand new Asus TUF Gaming F15 which is a gaming laptop, specifially to do some gaming party with friends on a TV, each one  using gamepads (around 4) and got this problem, it sucked big time, i confirm this laptop is using Intel AX 210

**Comment by Trihy** [comment]
Yes, sadly they refuse to recognize the problem. 

I don't have 4 controller to try, if you test 4 with other BT dongle, report and will add to the list.

**Comment by elliopitas** [comment]
same problem with intel killer 1650i
if multiple controllers are connected they work for some time but they keep disconnecting.

**Comment by Trihy** [comment]
Thanks for reporting. Added 1650 to the sh*tty adapter list.

https://github.com/Trihy/BT-Dongle-Compatibility-List

**Comment by BetaRavener** [comment]
Happy to finally find this thread after so many articles suggesting all kinds of crap. Same experience, PS4 controller loses connection seconds after connecting PS5 controller. Intel AX210 on motherboard but with nice cabled antennas to reduce interference. I'll also try posting on the Intel forums, but so far there has been no development on the story from Intel side, correct? Given that I have latest driver according their utility I guess not.

**Comment by Trihy** [comment]
There are many threads on intel wireless forum. But they couldnt care less.

Just avoid buying all AX trash.

**Comment by Nitheesh365** [comment]
> This is a classic bug with hidhide not configured properly;
> 1 update VIGEmBus https://github.com/ViGEm/ViGEmBus/releases
> 2 update hidhide https://github.com/ViGEm/HidHide/releases
> 3 reboot
> 4 run DS4windows as admin
> 5 configure hidhide (type it in cortana)
> 6 in applications tab you should have hidhidecliend.exe and DS4windows.exe
> 7 in devices tab put a lock on sony interactive entertainment BUT sony COMPUTER is fine or xbox 360... its depend of your emulation
> 8 close hidhide
> 9 in controllers tab of DS4windows if you see a key and NOT a x or check then you have fixed your problem
> 10 you may need to click stop start to see the key

Hi, I've the same issue. Tried this, but the Controllers Ex stays in X and the second controller keeps disconnecting after some time (can play for a bit). Any idea how to solve? Thank you for your time

**Comment by Trihy** [comment]
> > This is a classic bug with hidhide not configured properly;
> > 1 update VIGEmBus https://github.com/ViGEm/ViGEmBus/releases
> > 2 update hidhide https://github.com/ViGEm/HidHide/releases
> > 3 reboot
> > 4 run DS4windows as admin
> > 5 configure hidhide (type it in cortana)
> > 6 in applications tab you should have hidhidecliend.exe and DS4windows.exe
> > 7 in devices tab put a lock on sony interactive entertainment BUT sony COMPUTER is fine or xbox 360... its depend of your emulation
> > 8 close hidhide
> > 9 in controllers tab of DS4windows if you see a key and NOT a x or check then you have fixed your problem
> > 10 you may need to click stop start to see the key
> 
> Hi, I've the same issue. Tried this, but the Controllers Ex stays in X and the second controller keeps disconnecting after some time (can play for a bit). Any idea how to solve? Thank you for your time

Yes. Buy a better BT adapter.

**Comment by Nitheesh365** [comment]
> > > This is a classic bug with hidhide not configured properly;
> > > 1 update VIGEmBus https://github.com/ViGEm/ViGEmBus/releases
> > > 2 update hidhide https://github.com/ViGEm/HidHide/releases
> > > 3 reboot
> > > 4 run DS4windows as admin
> > > 5 configure hidhide (type it in cortana)
> > > 6 in applications tab you should have hidhidecliend.exe and DS4windows.exe
> > > 7 in devices tab put a lock on sony interactive entertainment BUT sony COMPUTER is fine or xbox 360... its depend of your emulation
> > > 8 close hidhide
> > > 9 in controllers tab of DS4windows if you see a key and NOT a x or check then you have fixed your problem
> > > 10 you may need to click stop start to see the key
> > 
> > 
> > Hi, I've the same issue. Tried this, but the Controllers Ex stays in X and the second controller keeps disconnecting after some time (can play for a bit). Any idea how to solve? Thank you for your time
> 
> Yes. Buy a better BT adapter.

UPDATE : Got myself a TP-LINK UB500 and two Dualsense controllers are working perfectly for hours. It's quite surprising considering I have Asus TUF DASH F15, supposedly with Bluetooth 5.2 😅. Either way, thank you for the response, it fixed the issues.

**Comment by Nitheesh365** [comment]
What fixed my issue was. Got a TP LINK UB500, disabled the built-in bluetooth (from device manager). Paired the 2X Dualsense controllers again to laptop. Everything working perfectly now, hours of gameplay without any issues so far.

**Comment by Trihy** [comment]
Great. Intel built in BT is trash level hardware.

**Comment by sebschlicht** [comment]
It's a relief to discover this thread right before you go mad and reinstall Windows just to get two BT controllers running smoothly.

If this is a bandwith problem, specifically occurring with Intel chips, wouldn't it be best to use whatever BT adapter is in the PS4 or PS5?

I just tried to find the BT adapter of the PS5 in some component lists in the web but didn't succeed. Does anyone have that information already?

**Comment by BetaRavener** [comment]
You can do that research, or save yourself time for actually playing something and get the TP-Link UB500 which many others and now me can confirm works.. 

And btw just 1 minute search out of curiosity, and while I don't have original part number, there are plenty of "compatible" PS5 BT modules being sold online. Reason? They seem to be often faulty (e.g. [thread here](https://www.reddit.com/r/PS5/comments/k4p9qf/dualsense_controller_bluetooth_interference/)). So the original chip is not always the best - I mean, right, Intel?

**Comment by Nitheesh365** [comment]
Yup. I was having same issue and can definitely confirm TP UB 500, works
for me. At least for 2 Dualsense controllers at the same time.

**Comment by Baton4ik48** [comment]
The funny thing is that I have Windows 11 and after reinstalling it, two gamepads stopped working at the same time. Before that, two DualSeanses worked together on WIndows 11, probably there were some correct drivers in another assembly.....

ps. I have Intel AX201

**Comment by Baton4ik48** [comment]
> The funny thing is that I have Windows 11 and after reinstalling it, two gamepads stopped working at the same time. Before that, two DualSeanses worked together on WIndows 11, probably there were some correct drivers in another assembly.....
> 
> ps. I have Intel AX201

lol, well, after all, this is not a matter of drivers, but of windows 11 updates,
I first installed another build of Windows 11 and two controllers worked for me, after updating Windows 11 and rebooting the PC, I started having problems again, the controllers only worked one at a time, they DID NOT WORK together anymore, I went home to get my flash drive and installed my Windows 11 which is not updated (tracker) and everything works fine on it (all both controllers are visible)

![123](https://github.com/Ryochan7/DS4Windows/assets/63069613/8adf38ff-361b-4210-b503-1466840a8055)

**Comment by Trihy** [comment]
The lazy ass at intel started working on improving this. 

Remember to use drivers from 23.xx.xx branch

https://www.intel.la/content/www/xl/es/download/18649/intel-wireless-bluetooth-for-windows-10-and-windows-11.html

Probably on upcoming builds will be improved a little more.

**Comment by Trihy** [comment]
With that branch, 2 controllers kind of works.

3 or 4 are still problematic with this shitty adapter.

**Comment by Baton4ik48** [comment]
> The lazy ass at intel started working on improving this.
> 
> Remember to use drivers from 23.xx.xx branch
> 
> https://www.intel.com/content/www/us/en/download/18649/intel-wireless-bluetooth-for-windows-10-and-windows-11.html
> 
> Probably on upcoming builds will be improved a little more.
> 
> BTW. Problem is intel, not windows. Windows update still uses old branch driver.

By the way, while I was writing to you, after 15 minutes everything broke again and the two joysticks no longer connect......
I installed BT-23.10.0-64UWD-Win10-Win11.exe using your link
doesn't work either......
what kind of dancing with a tambourine...

**Comment by Trihy** [comment]
If they don't connect, then you have another problem. Check your BT antenna.

**Comment by Baton4ik48** [comment]
> If they don't connect, then you have another problem. Check your BT antenna.

I freaked out and went and bought myself just TPLINK UB500

everything works perfectly with it...



[https://github.com/Trihy/BT-Dongle-Compatibility-List](url)

**Comment by Trihy** [comment]
Nice. That one works pretty good.

**Comment by Baton4ik48** [comment]
> Nice. That one works pretty good. And it´s cheap.

I'll still wait for Intel developers to figure out and fix this problem on Intel AX201 and other chips...It’s kind of a shame.....

**Comment by 1472583610** [comment]
Tl;dr version: 

Intel AX200 here (integrated in the ASRock X570 Taichi motherboard), driver 22.160.something. Fixed by reverting back to 22.150.0.6.

Long version:

I was just having the very same issue. Connected one DS, it stayed connected. Connected the 2nd one, the 1st got almost immediately disconnected, but the 2nd stayed on. When I reconnected the 1st one, the 2nd one got disconnected. This was very well reproducible.

I played and finished several co-op games with my wife, so I knew it wasn't a bandwidth issue. It just suddenly stopped working. Checked the driver version (Device Manager -> Bluetooth -> Intel(R) Wireless Bluetooth(R) -> Driver tab) and saw the button Roll Back Driver is active. I reverted the driver and both controllers now stay connected.

Speculation:

That said, if you can choose your BT adapter, I would heed the advice of others and try another adapter. The connection is not always stable. Possibly due to interference, but I don't have too many devices active (2 phones, a tablet and 2 watches) along with Wi-Fi (2 computers). Not sure how much a different adapter could help here, but it feels like BT should be able to handle significantly more devices around.

**Comment by Trihy** [comment]
It´s a BW problem or related to poll rate. It was partially resolved on some drivers. Is not a signal problem.

There is a thread on intel forums explaining all this.

22.150.0.6 Doesnt fix the issue. The issue is being worked on on intel BT driver 23.xx.xx and will have a proper solution or something like this on next quarter.

**Comment by 1472583610** [comment]
Yup, I can now confirm that the BT driver version is not the fix. The issue is back after 4 days of working without any problems while the BT driver is still on 22.150.0.6. Currently, I have no idea what changed and why the issue is back after days of use without any issues, not even with connection stability.

**Comment by Trihy** [comment]
Start by using the 23.xx.xx driver. You can download them on Intel site. A fix will come for that branch eventually.

**Comment by 1472583610** [comment]
I did already, updated and it works again. Quite confusing as to what exactly is the issue. If it didn't work at all, I'd be more "fine" with it, but I can't wrap my head around the fact that it works fine and then suddenly stops, with no obvious change.

**Comment by Trihy** [comment]
Yes for most users simply doesn't work.

But it's a know issue. If you need real support for more than one controller get a ub500 until this shitty BT gets properly fixed.
